### ISSUE TITLE
[IOS] app dialog for sharing files is blank

### ISSUE THREAD

**Issue by BlindHenry** [RocketChat/Rocket.Chat.ReactNative#2760]
### Description:

On IOS the app dialog for sharing files with contacts or channels is blank.

### Environment Information:

- Rocket.Chat Server Version: 3.9.4
- Rocket.Chat App Version: 4.13.0.22082
- Device Name: iPad Gen 7
- OS Version: 14.3

### Steps to reproduce:

1. go in any app, like photos or safari
2. press on share and select rocketchat
3. dialog is blank

### Expected behavior:

I'm able to use the share function to share files with rocketchat

### Actual behavior:

dialog is blank

**Comment by diegolmello** [comment]
Hey. I tested here and it worked. Can you share more info?
Maybe a video?

**Comment by BlindHenry** [comment]
> Hey. I tested here and it worked. Can you share more info?
> Maybe a video?

I have made a small screen capture for you, with the photo app trying to share a image.

https://user-images.githubusercontent.com/76985184/103648955-f53e8480-4f5d-11eb-8436-5093fc11c054.MOV

**Comment by BlindHenry** [comment]
Same on the last 4.13.1 version.

https://user-images.githubusercontent.com/76985184/103827576-9e2dd200-5079-11eb-8385-e5099f54cd28.MP4

**Comment by diegolmello** [comment]
@BlindHenry I'm going to fix it ASAP.
You said it happens from Safari.
A simple URL link raises the same issue?

**Comment by BlindHenry** [comment]
> @BlindHenry I'm going to fix it ASAP.
> You said it happens from Safari.
> A simple URL link raises the same issue?

@diegolmello Yes same on Safari with a simple URL (As an example I tried to share this URL: https://blog.fefe.de)

**Comment by BlindHenry** [comment]
Here as a video example

https://user-images.githubusercontent.com/76985184/103892223-81ca7d80-50eb-11eb-9e33-fc239ea72f43.MP4

**Comment by diegolmello** [comment]
Do you have any other device to try it out and check if it works?

**Comment by BlindHenry** [comment]
@diegolmello Yes, an iPhone X with iOS 14.2 
and rocket app version 4.12.0 that works, after an update to 4.13.1 it dosen't work and gives a blank dialog.

**Comment by diegolmello** [comment]
I've been trying my best to reproduce here, but it works all the time.
Can you try uninstalling and installing your app completely?

**Comment by deepLifer** [comment]
Hello, on my Iphone XR this bug also presents (((
RC Version - 4.13.1.22188

**Comment by BlindHenry** [comment]
> I've been trying my best to reproduce here, but it works all the time.
> Can you try uninstalling and installing your app completely?

Sorry this dosen‘t helped.
I already Fried this, this Problem is also präsent after a complete factory wipe, (only installieren Rocket from store Nottingham else)

**Comment by BlindHenry** [comment]
A friend tested with iPad Gen8 iOS 14.0 - RocketChat from store (never used before)

Without login to a rocket workspace the share dialog show the message:
![Bildschirm01](https://user-images.githubusercontent.com/76985184/103903912-d5919280-50fc-11eb-9cc0-22776ed57da6.png)
In english something like: No rocket server, you need at least once.
 
After joining open.rocket.chat, same as my device: the dialog is blank.

Is there any secret debug option that I can enable to help you? @diegolmello

**Comment by diegolmello** [comment]
@gerzonc was able to reproduce and he's taking a look on it already.
Thanks.

**Comment by diegolmello** [comment]
Ok, found it.
Memory is very limited for sharing files (app extension).
120 MB on some devices.
We'll have to analyse it.

**Comment by diegolmello** [comment]
@BlindHenry Can you [join TestFlight](https://testflight.apple.com/join/7I3dLCNT), download the latest version and try again?
It contains #2788.
It's the version 4.13.1.22299.

**Comment by BlindHenry** [comment]
@diegolmello 
Tested with iPad Gen 7 iOS 14.3 and it worked.

**Comment by diegolmello** [comment]
@BlindHenry Ok. I'll add it on the next version then.
Beta is planned to be released next Monday and public release on 25th.
Thanks.

**Comment by User6554** [comment]
Unfortunately I got the same issue as BlindHenry. The same blank page appears if I try to send an attachment trough Rocket.Chat mobile App. 

**The important thing is, though, that I downloaded the 4.14.0.22470 Beta via Testflight and had the same issue on an iPhone 12 (iOS 14.3) and an iPad Pro 2nd gen (iOS 14.3).** 

I am not sending any pictures or videos because it happens exactly as BlindHenry shows it with his recordings. 
Thanks a lot!

**Comment by ronakT-addweb** [comment]
> Unfortunately I got the same issue as BlindHenry. The same blank page appears if I try to send an attachment trough Rocket.Chat mobile App.
> 
> **The important thing is, though, that I downloaded the 4.14.0.22470 Beta via Testflight and had the same issue on an iPhone 12 (iOS 14.3) and an iPad Pro 2nd gen (iOS 14.3).**
> 
> I am not sending any pictures or videos because it happens exactly as BlindHenry shows it with his recordings.
> Thanks a lot!

**App version : 4.13.1.22188**
server version : 3.9.3

**Comment by diegolmello** [comment]
@ronakT-addweb can you try beta? https://github.com/RocketChat/Rocket.Chat.ReactNative/issues/2760#issuecomment-758987411

**Comment by ronakT-addweb** [comment]
> @ronakT-addweb can you try beta? [#2760 (comment)](https://github.com/RocketChat/Rocket.Chat.ReactNative/issues/2760#issuecomment-758987411)

Yup, It's working there.

**Comment by User6554** [comment]
@diegolmello, I would like to contribute somehow in solving the issue. I made this screen recording to highlight how this is happening to me, still with the new beta 4.14.0.22470:

https://user-images.githubusercontent.com/77775128/105392135-2637de00-5c1b-11eb-841c-cfb9c0975c1f.MP4

Tell me please if I can send any other information. Thanks.

**Comment by diegolmello** [comment]
@User6554 Some work was done here https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/2788
Share extensions have memory limit on iOS and vary depending on the device.
It's 120MB on most devices.
I'd appreciate a PR solving this issue.
Also, can you open a new issue?
Thanks!

**Comment by User6554** [comment]
@diegolmello Sure. I will open a new issue. What do you mean with "PR"? I am sorry, I am completely new to GitHub.

**Comment by diegolmello** [comment]
PR = Pull Request.
It's when you to send a code to add a new functionality or a fix to some issue.
I'm sorry, I thought you were a developer.

Open a new issue and I'll ask you to try the upcoming fix on TestFlight.
It might take a while for us to work on this again though.

**Comment by User6554** [comment]
@diegolmello No worries. Sure. I will open a new issue and gladly help you with the TestFlight tests. Thanks.

**Comment by diegolmello** [comment]
We just released 4.14.1 on App Store.
Google Play is taking longer to review it though.
After we land it there, we're going to resume the release process.
Thanks.

**Comment by User6554** [comment]
@diegolmello Thank you. I downloaded version 4.14.1.22787 from the AppStore on my iPhone 12 (256GB). Unfortunately, the issue persists. I restarted my phone hoping it would make a difference. Please tell me if you need any screenshot or screen recording. Thank you!

**Comment by diegolmello** [comment]
@User6554 We tested with a couple of users and it started to work.

The build was on TestFlight for a while.
I was hoping you tested it.
> @diegolmello No worries. Sure. I will open a new issue and gladly help you with the TestFlight tests. Thanks.

> Share extensions have memory limit on iOS and vary depending on the device.
> It's 120MB on most devices.
A deeper refactor is needed for it work if it's the case, so it might take a while for us to plan again.
Some apps do a heavy compression in order to share media through share extension and we can't do that.
We have to follow the settings on admin panel.
The best workaround right now is to open the app and share the file.

Can you create a new issue with details on how to reproduce it?
Give any details you have.
Thanks.

**Comment by User6554** [comment]
@diegolmello Yes, I think I saw, that for some users it started to work. 
I also tested the TestFlight release (4.14.1) and the new AppStore release (4.14.1) but unfortunately it did not work either.

Before I open a new issue: Can it be somehow related to the fact that I am connected to a self-hosted workspace. Everything else is working fine though.
If not, I will make Screenshots, make a screen recording and open a new issue. 

Thank you in advance,
Greeting, 
User6554
